ivy_tests/test_ivy/test_functional/test_core/test_gradients.py

```diff
@@ -348,11 +516,78 @@ def test_adam_update(ws_n_grads_n_lr_n_wsnew, dtype, tensor_fn, device, call):
     if call in [helpers.torch_call]:
         # pytorch scripting does not support internal function definitions
         return
+    
+#adam_update
+@given(
+    dtype_and_w=helpers.dtype_and_values(ivy_np.valid_float_dtypes),
+    as_variable=st.booleans(),
+    native_array=st.booleans(),
+    num_positional_args=helpers.num_positional_args(fn_name="adam_update"),
+    container=helpers.list_of_length(st.booleans(), 4),
+    instance_method=st.booleans(),
+    dcdw=st.floats(allow_infinity=False,allow_nan=False),
+    lr=st.floats(min_value=0.0,max_value=1.0,allow_nan=False),
+    mw_tm1=st.floats(allow_infinity=False,allow_nan=False),
+    vw_tm1=st.floats(allow_infinity=False,allow_nan=False),
+    step=st.floats(allow_infinity=False,allow_nan=False),
+    beta1=st.floats(min_value=0.0,max_value=1.0,allow_nan=False).filter(lambda x: x != 0),
+    beta2=st.floats(min_value=0.0,max_value=1.0,allow_nan=False).filter(lambda x: x != 0),
+    epsilon=st.floats(min_value=0.0,max_value=1.0,allow_nan=False).filter(lambda x: x != 0),
+    inplace=st.booleans(),
+    stop_gradients=st.booleans()
+)
+def test_adam_update(
+    dtype_and_w,
+    as_variable,
+    native_array,
+    num_positional_args,
+    container,
+    instance_method,
+    fw,
+    dcdw,
+    lr,
+    mw_tm1,
+    vw_tm1,
+    step,
+    beta1,
+    beta2,
+    epsilon,
+    inplace,
+    stop_gradients
+):
+    dtype, w = dtype_and_w
+    w = np.asarray(w,dtype=dtype)
+    dcdw = np.asarray(dcdw,dtype=dtype)
+    mw_tm1 = np.asarray(mw_tm1,dtype=dtype)
+    vw_tm1 = np.asarray(vw_tm1,dtype=dtype)
+    step = np.asarray(step,dtype=dtype)
+    helpers.test_function(
+        dtype,
+        as_variable,
+        False,
+        native_array,
+        fw,
+        num_positional_args,
+        container,
+        instance_method,
+        "adam_update",
+        w=w,
+        dcdw=dcdw,
+        lr=lr,
+        mw_tm1=mw_tm1,
+        vw_tm1=vw_tm1,
+        step=step,
+        beta1=beta1,
+        beta2=beta2,
+        epsilon=epsilon,
+        inplace=inplace,
+        stop_gradients=stop_gradients
+    )
 
 
 # layerwise_adam_update
-@pytest.mark.parametrize(
-    "ws_n_grads_n_lr_n_wsnew",
+@given(
```

Comment:
When using use hard-coded ground truth values (which is necessary for compositional functions), then we should use a unique name for the test, such as `test_<func_name>_ground_truth`, which should be added alongside the self-consistency test, which does not use any ground truth values.